file_browser/src/components/FolderView.tsx

```diff
@@ -0,0 +1,102 @@
+import { SystemItem } from "../classes/SystemItem";
+import { Folder } from "../classes/Folder";
+import { observer } from "mobx-react-lite";
+import { useMemo, useState } from "react";
+import fileSystemStore from "../stores/FileSystemStore";
+import { FileView } from "./FileView";
+import { FileClass } from "../classes/File";
+import s from "./FolderView.module.scss";
+
+// icons
+import openedIcon from "../assets/openedFolder.png";
+import whiteOpenedIcon from "../assets/openedFolderWhite.png";
+import closedIcon from "../assets/closedFolder.png";
+import whiteClosedIcon from "../assets/closedFolderWhite.png";
+import leftArrow from "../assets/chevron.png";
+import whiteLeftArrow from "../assets/whiteLeftArrow.png";
+import downArrow from "../assets/down-arrow.png";
+import whiteDownArrow from "../assets/whiteDownArrow.png";
+
+const FolderView = observer(({ folder }: { folder: Folder }) => {
+  const [isFolderOpened, setIsFolderOpened] = useState(false);
+
+  const folderIcon = useMemo(() => {
+    if (isFolderOpened && folder.files.size !== 0) {
+      if (fileSystemStore.selectedId === folder.id) {
+        return whiteOpenedIcon;
+      } else return openedIcon;
+    } else {
+      if (fileSystemStore.selectedId === folder.id) {
+        return whiteClosedIcon;
+      } else return closedIcon;
+    }
+  }, [
+    isFolderOpened,
+    folder.files.size,
+    fileSystemStore.selectedId,
+    folder.id,
+  ]);
+
+  const arrowIcon = useMemo(() => {
+    if (isFolderOpened) {
+      if (fileSystemStore.selectedId === folder.id) {
+        return whiteDownArrow;
+      } else return downArrow;
+    } else {
+      if (fileSystemStore.selectedId === folder.id) {
+        return whiteLeftArrow;
+      } else return leftArrow;
+    }
+  }, [isFolderOpened, fileSystemStore.selectedId, folder.id]);
+
+  const handleSelectItem = (item: SystemItem) => {
+    fileSystemStore.setSelected(item);
+  };
+
+  return (
+    <>
+      {folder.name.toLowerCase().includes(fileSystemStore.searchText) && (
```

Comment:
You don't need this anymore - the filtering is done on the data layer after this update.